### ISSUE TITLE
Question: Is `helm test --logs` incompatible with hook delete policies?

### ISSUE THREAD

**Issue by mdblr** [helm/helm#8018]
The output of `helm version`:
```
version.BuildInfo{Version:"v3.1.2", GitCommit:"d878d4d45863e42fd5cff6743294a11d28a9abce", GitTreeState:"clean", GoVersion:"go1.13.8"}
```
Output of `kubectl version`:

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
N/A

I am running `helm test --logs ....`. I have set a "helm.sh/hook-delete-policy" of hook-succeeded,hook-failed in my tests. According to the Helm docs, the logs flag will "Dump the logs from test pods (this runs after all tests are complete, but before any cleanup)". I believed the delete policy was part of the cleanup, but I hit the error below:

```
Error: unable to get pod logs for hello-world-1231321-foo-bar: pods "hello-world-1231321-foo-bar" not found
```

Are these two features compatible? Is there a way to get the logs using Helm and automatically delete the test pods?

Thanks

**Comment by aweis89** [comment]
Running into this issue as well

**Comment by hottomato4** [comment]
+1

**Comment by bacongobbler** [comment]
In the interest of preventing spam to users subscribing to this thread, please add a :+1: to the OP instead of commenting "+1" or “me too!”. That way we can keep the conversation relevant to the discussion at hand. I've gone ahead and deleted those comments. Thanks!

**Comment by foosinn** [comment]
@bacongobbler: I run into the same issue without having a `helm.sh/hook-delete-policy` specified. 

Iwas able to fix it using a Pod resource for my tests instead of a Job.

**Comment by 11399371** [comment]
Hi @mdblr @bacongobbler , is there a fix/wa for this? 
I run into the same issue with helm3, but helm2 is able to collect the the logs.
My only WA is to explicitly delete the pod in an external script.

**Comment by tejaswiniVadlamudi** [comment]
@bacongobbler : Logs are not displayed if hook deletion policies are in place irrespective of test implementation object carrier (pod or a k8s job).

**Comment by lindhe** [comment]
The PR seems to be stuck at writing unit tests. Can someone help with that?

**Comment by rdesgroppes** [comment]
@cndoit18 that would be amazing!

**Comment by cndoit18** [comment]
> I will try to fix this thing and #9264

Sorry, I saw that someone was already working on it, so I didn't follow up
